### PR TITLE
Add stories for SearchResultList component

### PR DESCRIPTION
**Pull request by mrblippy**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included

##### Description of change

#1306 

This PR adds three stories for the SearchResultList component:
- No results found
- Basic list of results
- Loading results page (with spinner)

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by edalex-ian** [comment]
Heya @mrblippy 

So now that we have Storybook 6 over here, how come you went old skool on this (knobs and actions)?

**Comment by mrblippy** [comment]
> Heya @mrblippy
> 
> So now that we have Storybook 6 over here, how come you went old skool on this (knobs and actions)?

This PR https://github.com/openequella/openEQUELLA/pull/2182 has the initial prep-work. Including adding the "@storybook/addon-essentials" addon, which allows for the Storybook Controls functionality to be used

File: Source/Plugins/Core/com.equella.core/js/.storybook/main.js [hunk 1]
```diff
@@ -2,9 +2,7 @@ module.exports = {
   stories: ["../__stories__/**/*.stories.tsx"],
   addons: [
     "@storybook/addon-a11y",
+    "@storybook/addon-essentials",
     "@storybook/addon-actions",
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/js/.storybook/main.js:6]: the difference between this and
https://github.com/openequella/openEQUELLA/blob/03242c2cee9e47ad4a0c5fb02b72e37b0ce85dab/Source/Plugins/Core/com.equella.core/js/.storybook/main.ts#L4-L5

could cause merge conflicts down the road.
If possible consider merging develop into this branch again.

File: Source/Plugins/Core/com.equella.core/js/package.json [hunk 2]
```diff
@@ -65,16 +65,13 @@
   "license": "Apache-2.0",
   "devDependencies": {
     "@babel/core": "7.11.1",
+    "@storybook/addon-a11y": "6.0.12",
+    "@storybook/addon-actions": "^6.0.17",
+    "@storybook/addon-essentials": "6.0.13",
     "@storybook/preset-typescript": "3.0.0",
+    "@storybook/react": "6.0.12",
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/js/package.json]: another place where these changes may conflict with
https://github.com/openequella/openEQUELLA/blob/815d1037cc7dfe2079c3344384ff437eac9884bd/Source/Plugins/Core/com.equella.core/js/package.json#L74-L76

from `develop`.
merging branches may make things easier long term.

File: Source/Plugins/Core/com.equella.core/js/__stories__/search/SearchResultList.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  component: SearchResultList,
+  argTypes: {
+    onClearSearchOptions: { action: "onClearSearchOptions called" },
+  },
+} as Meta<SearchResultListProps>;
+
+const sharedPaginationArgs = {
+  currentPage: defaultSearchOptions.currentPage,
+  rowsPerPage: defaultSearchOptions.rowsPerPage,
+  /* I wasn't able to get nested actions to work inside argTypes, so falling back to the old addon-actions for these. See https://github.com/storybookjs/storybook/issues/11525 and https://github.com/storybookjs/storybook/issues/10979#issuecomment-657640744*/
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/js/__stories__/search/SearchResultList.stories.tsx:42]: :man_shrugging: 
an alternative is to use desctucturing and assignment.
flattening the arg types to a single level, then recreating the multi-level structure.
a similar-ish example:
https://github.com/openequella/openEQUELLA/blob/815d1037cc7dfe2079c3344384ff437eac9884bd/Source/Plugins/Core/com.equella.core/js/__stories__/components/SettingsListControl.stories.tsx#L52-L73

File: Source/Plugins/Core/com.equella.core/js/.storybook/main.js (hunk 1 above)
Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/js/.storybook/main.js:6]: Ah yeah, good point we've got @ChristianMurphy 's WIP in now.

Let's merge `develop` to component again, and then merge that to your branch @mrblippy . You cool with that?

File: Source/Plugins/Core/com.equella.core/js/__stories__/search/SearchResultList.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  searchResultItems: singlePageSearch.results,
+  showSpinner: true,
+  paginationProps: {
+    ...sharedPaginationArgs,
+    count: singlePageSearch.available,
+  },
+  orderSelectProps: {
+    ...sharedOrderSelectProps,
+  },
+};
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/js/__stories__/search/SearchResultList.stories.tsx:99]: Would this be the same as:

```suggestion
LoadingSearchResultListComponent.args = {
  ...BasicSearchResultListComponent.args,
  showSpinner: true,
};
```

Mind you, I also wonder if in `BasicSearchResultListComponent.args` you could also mostly spread from `EmptyResultListComponent.args`. Thoughts?

File: Source/Plugins/Core/com.equella.core/js/.storybook/main.js (hunk 1 above)
Comment (mrblippy) [reply Source/Plugins/Core/com.equella.core/js/.storybook/main.js:6]: Yes, i can certainly do that

**Review by edalex-ian — COMMENTED** [review]
Good stuff @mrblippy . Any chance we could just try a bit more re-use with the Args?

Thanks for battling through with the nested actions issue. I know you tried several approaches including the `Pick` etc. I guess we're enjoying riding the new wave. ;)

File: Source/Plugins/Core/com.equella.core/js/__stories__/search/SearchResultList.stories.tsx (hunk 4 above)
Comment (mrblippy) [reply Source/Plugins/Core/com.equella.core/js/__stories__/search/SearchResultList.stories.tsx:99]: Yeah, i think that makes sense. I will give it a go

File: Source/Plugins/Core/com.equella.core/js/package.json [hunk 5]
```diff
@@ -66,12 +66,12 @@
   "devDependencies": {
     "@babel/core": "7.11.4",
     "@storybook/preset-typescript": "3.0.0",
+    "@storybook/react": "6.0.16",
     "@testing-library/jest-dom": "5.11.2",
     "@testing-library/react": "10.4.7",
     "@storybook/addon-a11y": "6.0.16",
     "@storybook/addon-essentials": "6.0.16",
     "@storybook/addon-knobs": "6.0.17",
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/js/package.json:74]: there's a mix of version 6.0.16 and 6.0.17 here for storybook.
does version 6.0.17 exist for all the storybook packages?

File: Source/Plugins/Core/com.equella.core/js/package.json [hunk 6]
```diff
@@ -66,12 +66,12 @@
   "devDependencies": {
     "@babel/core": "7.11.4",
     "@storybook/preset-typescript": "3.0.0",
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/js/package.json:68]: this looks like this looks like it is synced with `component/new-search-page` but still missing some changes from `develop`.
preset-typescript is no longer needed with storybook 6.

https://github.com/openequella/openEQUELLA/blob/815d1037cc7dfe2079c3344384ff437eac9884bd/Source/Plugins/Core/com.equella.core/js/package.json#L74-L76

**Review by edalex-ian — APPROVED** [review]
I'm happy with it, but looks like @ChristianMurphy has found maybe an issue with the merges. Better check/fix that first.

File: Source/Plugins/Core/com.equella.core/js/package.json (hunk 6 above)
Comment (mrblippy) [reply Source/Plugins/Core/com.equella.core/js/package.json:68]: Thanks for pointing this out Christian. I will double check these issues before i merge

File: Source/Plugins/Core/com.equella.core/js/__stories__/search/SearchResultList.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+export const LoadingSearchResultListComponent: Story<SearchResultListProps> = (
+  args
+) => <SearchResultList {...args}></SearchResultList>;
+
+LoadingSearchResultListComponent.args = {
+  ...BasicSearchResultListComponent.args,
+  showSpinner: true,
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/js/__stories__/search/SearchResultList.stories.tsx:83]: This is the goods. This is hard to test with React Testing Library as it's all kind of real time. But we have an enzyme test which tests it based on stepping through with `component.update()`.

Now that I know this is at least covered here, I'll remove the test and stop trying to refactor it. Thanks @mrblippy !

**Pull request merged**